isic/core/tests/test_search.py

```diff
@@ -73,69 +73,67 @@ def collection_with_image(search_index, image_factory, collection_factory):
 
 
 @pytest.mark.django_db
-def test_core_api_image_search_private_image_as_guest(private_searchable_image, api_client):
-    r = api_client.get("/api/v2/images/search/")
-    assert r.status_code == 200, r.data
-    assert r.data["count"] == 0, r.data
+def test_core_api_image_search_private_image_as_guest(private_searchable_image, client):
+    r = client.get("/api/v2/images/search/")
+    assert r.status_code == 200, r.json()
+    assert r.json()["count"] == 0, r.json()
 
 
 @pytest.mark.django_db
-def test_core_api_image_search_images_as_guest(searchable_images, api_client):
-    r = api_client.get("/api/v2/images/search/")
-    assert r.status_code == 200, r.data
-    assert r.data["count"] == 1, r.data
+def test_core_api_image_search_images_as_guest(searchable_images, client):
+    r = client.get("/api/v2/images/search/")
+    assert r.status_code == 200, r.json()
+    assert r.json()["count"] == 1, r.json()
 
 
 @pytest.mark.django_db
-def test_core_api_image_search_contributed(
-    private_searchable_image, authenticated_api_client, user
-):
+def test_core_api_image_search_contributed(private_searchable_image, authenticated_client, user):
     private_searchable_image.accession.cohort.contributor.owners.add(user)
     add_to_search_index(private_searchable_image)
     get_elasticsearch_client().indices.refresh(index="_all")
 
-    r = authenticated_api_client.get("/api/v2/images/search/")
-    assert r.status_code == 200, r.data
-    assert r.data["count"] == 1, r.data
+    r = authenticated_client.get("/api/v2/images/search/")
+    assert r.status_code == 200, r.json()
+    assert r.json()["count"] == 1, r.json()
 
 
 @pytest.mark.django_db
 def test_core_api_image_search_shares(
-    private_searchable_image, authenticated_api_client, user, staff_user
+    private_searchable_image, authenticated_client, user, staff_user
 ):
     private_searchable_image.shares.add(user, through_defaults={"creator": staff_user})
     private_searchable_image.save()
     add_to_search_index(private_searchable_image)
     get_elasticsearch_client().indices.refresh(index="_all")
 
-    r = authenticated_api_client.get("/api/v2/images/search/")
-    assert r.status_code == 200, r.data
-    assert r.data["count"] == 1, r.data
+    r = authenticated_client.get("/api/v2/images/search/")
+    assert r.status_code == 200, r.json()
+    assert r.json()["count"] == 1, r.json()
 
 
 @pytest.mark.django_db
 @pytest.mark.parametrize("route", ["images/search/", "images/facets/"])
-def test_core_api_image_search_invalid_query(route, searchable_images, authenticated_api_client):
-    r = authenticated_api_client.get(f"/api/v2/{route}", {"query": "age_approx:[[[[]]]]"})
-    assert r.status_code == 400, r.data
+def test_core_api_image_search_invalid_query(route, searchable_images, authenticated_client):
+    r = authenticated_client.get(f"/api/v2/{route}", {"query": "age_approx:[[[[]]]]"})
+    assert r.status_code == 422, r.json()
```

Comment:
@danlamanna please take note of this change, I want to make sure this is not considered API-breaking.

Reply:
This is one of the very few places it matters: https://github.com/imagemarkup/isic-cli/blob/master/isic_cli/cli/types.py#L17-L18.

We should keep the error format for this endpoint. After this is merged I'll modify the CLI to use a custom endpoint for validating the search query string. After we wait a bit, we can change the server side.

Reply:
This has been addressed, but unfortunately I couldn't find a simpler way to do it than [f8d7eb4](https://github.com/ImageMarkup/isic/pull/745/commits/f8d7eb4bfd645da0311c8724a863b7613ceeb8a0).